aiida/tools/importexport/common/archive.py

```diff
@@ -311,37 +363,33 @@ def extract_tar(infile, folder, nodes_export_subfolder=None, silent=False):
     try:
         with tarfile.open(infile, 'r:*', format=tarfile.PAX_FORMAT) as handle:
 
-            try:
-                handle.extract(path=folder.abspath, member=handle.getmember('metadata.json'))
-            except KeyError:
-                raise CorruptArchive('required file `metadata.json` is not included')
+            if len(handle.getmembers()) == 1 and handle.getmembers()[0].size == 0:
+                raise CorruptArchive('no files detected in archive')
 
-            try:
-                handle.extract(path=folder.abspath, member=handle.getmember('data.json'))
-            except KeyError:
-                raise CorruptArchive('required file `data.json` is not included')
+            file_iterator = get_file_iterator(file_handle=handle, folderpath=folder.abspath, silent=silent, **kwargs)
 
-            if not silent:
-                print('EXTRACTING NODE DATA...')
-
-            for member in handle.getmembers():
+            for member in file_iterator:
                 if member.isdev():
                     # safety: skip if character device, block device or FIFO
                     print('WARNING, device found inside the import file: {}'.format(member.name), file=sys.stderr)
                     continue
                 if member.issym() or member.islnk():
                     # safety: in export, I set dereference=True therefore
                     # there should be no symbolic or hard links.
-                    print('WARNING, link found inside the import file: {}'.format(member.name), file=sys.stderr)
+                    print('WARNING, symlink found inside the import file: {}'.format(member.name), file=sys.stderr)
                     continue
                 # Check that we are only exporting nodes within the subfolder!
                 # TODO: better check such that there are no .. in the
                 # path; use probably the folder limit checks
                 if not member.name.startswith(nodes_export_subfolder + os.sep):
                     continue
+
+                update_description(member.name, file_iterator)
+
                 handle.extract(path=folder.abspath, member=member)
     except tarfile.ReadError:
-        raise ValueError('The input file format for import is not valid (1)')
+        raise ValueError('The input file format for import is not valid (not a tar file)')
+    close_progress_bar(leave=False)
```

Comment:
I appreciate that this is not an easy problem, but I am really a bit worried about the tight coupling of all these various components to the progress bar. You may have removed the necessity of passing an instance around, but just hiding this dependency by going through a global singleton is not actually decoupling it. Why do you need the progress bar to be updated here inside the archive class? Why can't this be done in the caller?

Is it really not possible to have the top import method instantiate a progress bar and update it when necessary. If it needs information that are hidden in lower lying functions/classes, rather than passing the progress bar in there, one could maybe implement a method to get the information needed. The import method could call that first, update the progress bar and then call the actual method. Can you point me to the code why you think it is crucial to have the `Archive` class do it?